examples/fuzzed_data_provider_fuzz_test.cc

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// A fuzz target that demonstrates the use of FuzzedDataProvider.
+
+#include <fuzzer/FuzzedDataProvider.h>
+#include <cstdint>
+#include <cstddef>
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
+    FuzzedDataProvider fuzzed_data(Data, Size);
+
+    // Intentionally using uint16_t here to avoid empty |second_part|.
+    size_t first_part_size = fuzzed_data.ConsumeIntegral<uint16_t>();
```

Comment:
Use `auto` here, since the type is already specified in ConsumeIntegral<>. Even better, make this `const auto`, since the value is not expected to change.